### DISCUSSION TITLE
location of config file

### DISCUSSION THREAD

**Discussion by goyalyashpal** [wezterm/wezterm#2909] [Q&A] [answered]
i can't find my wezterm config file! is there some command to open wezterm's config file, or to echo its location? or smth like that?

i had this issue here too: https://github.com/wez/wezterm/issues/2294#issuecomment-1201456967

![image](https://user-images.githubusercontent.com/19423063/210167577-65c8253a-0bcb-45a5-9d2b-ede1857b2d65.png)

i am on windows 10 20H2

**Comment by yutkat** [comment]
Look here. https://wezfurlong.org/wezterm/config/files.html

I think you can create wezterm.lua in the same folder as wezterm.exe, or you can set the specified environment variables and put the file there.

(BTW, I'm a Linux user, so I'm not checking.)

**Reply by goyalyashpal** [reply]
the config file is already created - as it contains the address of the shell. but i don't know where did i put it.... it was quite a lot of months ago

and no, it ain't coming from `%COMSPEC%` environment variable
https://wezfurlong.org/wezterm/config/launch.html#on-windows-systems

**Reply by wez** [reply, marked as answer]
Run `wezterm start cmd`, to explicitly launch `cmd.exe`, then when the window appears, press `CTRL-SHIFT-L` to open the debug overlay, then type `wezterm.config_file` and press enter; it should show the config file path that is in use.

https://wezfurlong.org/wezterm/config/lua/wezterm/config_file.html

**Reply by cebaa** [reply]
@wez Is debug overlay documented somewhere? If not, I feel like this is quite a useful feature that probably deserves its own page - or perhaps a page dedicated to "discovery and troubleshooting", so that it includes things like logging, debugging & such. It might help a lot of folks and lower the amount of silly questions you get :)

**Reply by wez** [reply]
It's documented at https://wezfurlong.org/wezterm/config/lua/keyassignment/ShowDebugOverlay.html and the bug report form has instructions on using it as part of the bug report submission.  In the macOS nightly build, you can see the debug overlay in the Help menu, and in the nightly in general, the new command palette (CTRL-SHIFT-P; mentioned in the changelog, but not yet fully written up in the main body of the docs) includes it as well.

I don't necessarily agree that adding more documentation will help reduce the quantity or improve the quality of questions: there are many questions that people ask that are answered exactly with examples in the documentation, but which the person asking did not search for.

That said, I do agree that having the debug overlay called out a bit more explicitly in a couple of places couldn't hurt anyone. Are there specific pages in the docs where you would you have appreciated seeing it?

**Reply by cebaa** [reply]
> It's documented at https://wezfurlong.org/wezterm/config/lua/keyassignment/ShowDebugOverlay.html 

Yeah, that's a bit hard to find for newcomers...

> bug report form has instructions on using it as part of the bug report submission.

Ah, OK, that's a good thing then. I guess I went the route of discussions first so I did not get the benefit of that.

> I don't necessarily agree that adding more documentation will help reduce the quantity or improve the quality of questions: there are many questions that people ask that are answered exactly with examples in the documentation, but which the person asking did not search for.

Yeah that's always a hard one. It's both that people in general tend not to bother searching, but also that you need to anticipate what they are searching for, which is hard in itself (as evidenced by my other discussion entry https://github.com/wez/wezterm/discussions/2970#discussioncomment-4718861).

> Are there specific pages in the docs where you would you have appreciated seeing it?

IMO a top-level entry - and very close to the top, perhaps after 4. Configuration - named something like "Troubleshooting" would give the best benefit - and I'd include all easy ways people could get more information (debugging, logging, profiling, etc.).

Obviously you know more about the question / discussion patterns to judge whether this will provide enough benefit to warrant it, but I'd assume having 5. Troubleshooting is better than having 5. Scrollback on average.

**Reply by goyalyashpal** [reply]
> Run `wezterm start cmd`, to explicitly launch `cmd.exe`, then when the window appears, press `CTRL-SHIFT-L` to open the debug overlay, then type `wezterm.config_file` and press enter; it should show the config file path that is in use.

```console
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
> wezterm.config_file
""
>
```

it showed these two empty quotes

**Reply by wez** [reply]
@yashpalgoyal1304 do you have `WEZTERM_CONFIG_FILE` set in the environment? Try unsetting it?

**Reply by goyalyashpal** [reply]
uh oh! sorry, I just checked the environment variables again. The `ComSpec` variable is defined at two places 🤦 

sorry for the trouble, and thanks a lot for cooperation and help.

**Comment by akash942** [comment]
(for Windows) from what I understood from the docs there is no default config file, you have to create it yourself. So just create a file named "wezterm.lua" in the same folder as wezterm.exe.
note: name the file as "wezterm.lua" and not ".wezterm.lua" which is given in docs, which I think is wrong for some reason because as soon as I removed the starting dot from the name it worked

**Reply by wez** [reply]
The docs say to create the config file in your home directory. This applies on all platforms.
https://wezfurlong.org/wezterm/config/files.html#quick-start

**Reply by akash942** [reply]
Sorry! Just found out that home directory means the one with my username like "C:\Users\yourusername" I was placing the lua file in "C:" and wondering why it didn't work 
Thanks for the reply!!!

**Reply by goyalyashpal** [reply]
hi @akash942 , these are also available via system variables.

u can see the list of predefined system variables in
* a system setting panel called "Environment Variables", or
* in doublecommander file explorer

u can directly use and access those variables in:
* windows explorer's address bar, or `cmd` shell with `%<var-name>%` syntax, 
* bash/git-bash with `$<var-name-case-sensitive>`
* powershell with `$env:<var-name>` syntax

so, for concrete example here, type `%userprofile%` in address bar of windows explorer and press enter -> the same path u discovered above.

**Comment by iilyak** [comment]
I am having the same issue. It is very confusing. I do have the `wizterm.lua` (and I did try the `.wizterm.lua` as well) in the `C:\Users\iilyak` folder. Nevertheless I have empty value for wezterm.config_file

```
Enter lua statements or expressions and hit Enter.
Press ESC or CTRL-D to exit
> wezterm.config_file
""
```


```
C:\Users\iilyak>dir
 Volume in drive C is OS
 Volume Serial Number is 28FC-B974

 Directory of C:\Users\iilyak

2024-09-29  11:17 AM    <DIR>          .
2023-07-23  11:00 AM    <DIR>          ..
........
2024-09-29  11:11 AM               262 wizterm.lua
              12 File(s)        104,698 bytes
              52 Dir(s)  141,035,696,128 bytes free
```

The `%userprofile%` resolves into `C:\Users\iilyak>`.

**Reply by iilyak** [reply]
In my case it was a typo `wizterm` -> `weztrerm`
